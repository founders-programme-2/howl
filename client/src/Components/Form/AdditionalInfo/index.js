import React, { Fragment } from 'react';
import {
  withStyles,
  FormControl,
  Input,
  InputLabel,
  MenuItem,
  ListItemText,
  Select,
  Checkbox,
  Chip,
  FormControlLabel,
  TextField,
} from '../../muIndex';
import { styles, MenuProps } from './muiStyles';
import { ErrMsg, ErrMsgInfoTrue } from './AdditionalInfo.style';
import { tags as arrTags, categories as arrCategory } from '../../data.json';

const AdditionalInfo = ({
  classes,
  tags,
  tagsChange,
  category,
  dropdownChange,
  infoTrue,
  checkboxChange,
  textChange,
  wlmConnection,
  additionalComments,
  tagsErr,
  categoryErr,
  infoTrueErr,
  wlmConnectionErr,
  additionalCommentsErr,
}) => {
  return (
    <Fragment>
      {categoryErr ? (
        <ErrMsg>{categoryErr}</ErrMsg>
      ) : (
        <p>
          Please select one major category under which your contribution falls
          under. We understand that there might be more than one fits, please
          select one and you will be given a chance to select some subcategories
          later.
        </p>
      )}
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="category" className={classes.label}>
          Category (required)
        </InputLabel>
        <Select
          aria-label="category"
          value={category}
          onChange={dropdownChange}
          inputProps={{
            name: 'category',
            id: 'category',
          }}
          disableUnderline
          className={classes.select}
        >
          {arrCategory.sort().map(cat => (
            <MenuItem value={cat} key={cat}>
              {cat}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
      {tagsErr ? (
        <ErrMsg>{tagsErr}</ErrMsg>
      ) : (
        <p>
          Please select one or more &#39;tags&#39; for your contribution. A tag
          can be thought of as a related theme or subcategory.{' '}
        </p>
      )}
      <FormControl
        className={`${classes.formControl} ${classes.formControlTags}`}
      >
        <InputLabel htmlFor="selectMultipleCheckbox" className={classes.label}>
          Tags (required)
        </InputLabel>
        <Select
          aria-label="tags"
          multiple
          value={tags}
          onChange={tagsChange}
          input={<Input id="selectMultipleCheckbox" />}
          disableUnderline
          className={classes.select}
          renderValue={selected => (
            <div className={classes.chips}>
              {selected.map(value => (
                <Chip key={value} label={value} className={classes.chip} />
              ))}
            </div>
          )}
          MenuProps={MenuProps}
        >
          {arrTags.sort().map(tag => (
            <MenuItem key={tag} value={tag}>
              <Checkbox checked={tags.indexOf(tag) > -1} />
              <ListItemText primary={tag} />
            </MenuItem>
          ))}
        </Select>
      </FormControl>
      {wlmConnectionErr ? (
        <ErrMsg>{wlmConnectionErr}</ErrMsg>
      ) : (
        <p>
          Tell us about yourself. What is your connection to the Women&#39;s
          Liberation Movement? Please do not share any private information here
          such as your phone number or address.
        </p>
      )}
      <TextField
        id="standard-multiline-static"
        label="Connection to Womenâ€™s Liberation Movement (required)"
        name="wlmConnection"
        value={wlmConnection}
        multiline
        rows="4"
        fullWidth
        placeholder="examples: activist, daughter of an activist, et cetera)"
        className={`${classes.textField} ${classes.textFieldWlm}`}
        margin="normal"
        onChange={textChange('wlmConnection')}
        InputProps={{
          disableUnderline: true,
        }}
        InputLabelProps={{
          className: classes.label,
        }}
      />
      {additionalCommentsErr ? (
        <ErrMsg>{additionalComments}</ErrMsg>
      ) : (
        <p>Anything else you would like to tell us?</p>
      )}
      <TextField
        id="additionalComments"
        label="Additional comments? (optional)"
        name="additionalComments"
        value={additionalComments}
        multiline
        rows="4"
        placeholder="Talk to us!"
        className={classes.textField}
        margin="normal"
        onChange={textChange('additionalComments')}
        InputProps={{
          disableUnderline: true,
        }}
        InputLabelProps={{
          className: classes.label,
        }}
      />
      <FormControl className={classes.formControlCheckbox}>
        {infoTrueErr ? <ErrMsgInfoTrue>{infoTrueErr}</ErrMsgInfoTrue> : null}
        <FormControlLabel
          className={classes.formControlLabel}
          control={
            <Checkbox
              checked={infoTrue}
              onChange={checkboxChange('infoTrue')}
              value="infoTrue"
            />
          }
          label="I certify that this information is true to the best of my knowledge and contains no personal attacks."
        />
      </FormControl>
    </Fragment>
  );
};

export default withStyles(styles)(AdditionalInfo);
