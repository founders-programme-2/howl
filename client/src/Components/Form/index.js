import React, { Component } from 'react';
import axios from 'axios';
import ContactInfo from './ContactInfo';
import Details from './Details';
import Post from './Post';
import AdditionalInfo from './AdditionalInfo';
import { MuiThemeProvider } from '../muIndex';
import { FormTheme, CntrForm } from './Form.style';
import Buttons from './Buttons';

class Form extends Component {
  state = {
    radio: 'textPost',
    tags: [],
    category: '',
    infoTrue: false,
    name: '',
    email: '',
    phone: '',
    location: '',
    selectedDate: new Date(),
    year: '',
    month: '',
    title: '',
    details: '',
    imageCap: '',
    imgPermission: false,
    imgLink: '',
    wlmConnection: '',
    additionalComments: '',
    message: '',
    uploadFlag: false,
    imageFlag: true,
  };

  radioChange = event => {
    this.setState({ radio: event.target.value });
  };

  tagsChange = event => {
    this.setState({ tags: event.target.value });
  };

  dropdownChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  checkboxChange = name => event => {
    this.setState({ [name]: event.target.checked });
  };

  textChange = name => event => {
    this.setState({ [name]: event.target.value });
  };

  handleDateChange = date => {
    const options = {
      year: 'numeric',
      month: 'long',
    };
    const dateModified = date.toLocaleDateString('en-US', options);
    const splitted = dateModified.split(' ');
    this.setState({
      selectedDate: dateModified,
      month: splitted[0],
      year: splitted[1],
    });
  };

  handleUploadFile = event => {
    const img = new FormData();
    img.append('uploadedImg', event.target.files[0]);
    this.setState({ uploadFlag: true, imageFlag: false }, () => {
      axios.post('/upload', img).then(({ data }) => {
        if (data.success) {
          this.setState({
            uploadFlag: false,
            imgLink: data.image,
            message: 'Image uploaded successfully!',
          });
        } else {
          this.setState({
            uploadFlag: false,
            imageFlag: true,
            imgLink: 'no url was retrieved',
            message: 'Something went wrong, please try again.',
          });
        }
      });
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    const submittedData = { ...this.state };
    axios.post('/posts/create', submittedData).then(() => {
      const { history } = this.props;
      history.push('/story');
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    const submittedData = { ...this.state };
    axios.post('/posts/create', submittedData).then(res => {
      const { history } = this.props;
      history.push('/story');
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    const submittedData = { ...this.state };
    axios.post('/posts/create', submittedData).then(() => {
      const { history } = this.props;
      history.push('/story');
    });
  };

  render() {
    const {
      radio,
      tags,
      category,
      infoTrue,
      name,
      email,
      phone,
      location,
      selectedDate,
      title,
      details,
      imageCap,
      imgPermission,
      wlmConnection,
      additionalComments,
      message,
      imgLink,
      uploadFlag,
      imageFlag,
    } = this.state;
    return (
      <MuiThemeProvider theme={FormTheme}>
        <main>
          <header>
            <h1>Add to HOWL</h1>
          </header>
          <CntrForm>
            <ContactInfo
              name={name}
              email={email}
              phone={phone}
              textChange={this.textChange}
            />
            <Details
              radio={radio}
              radioChange={this.radioChange}
              location={location}
              selectedDate={selectedDate}
              textChange={this.textChange}
              handleDateChange={this.handleDateChange}
              dropdownChange={this.dropdownChange}
            />
            <Post
              title={title}
              details={details}
              imageCap={imageCap}
              imgPermission={imgPermission}
              checkboxChange={this.checkboxChange}
              textChange={this.textChange}
              handleUploadFile={this.handleUploadFile}
              radio={radio}
              imgLink={imgLink}
              message={message}
              uploadFlag={uploadFlag}
              imageFlag={imageFlag}
            />
            <AdditionalInfo
              tags={tags}
              category={category}
              infoTrue={infoTrue}
              tagsChange={this.tagsChange}
              dropdownChange={this.dropdownChange}
              checkboxChange={this.checkboxChange}
              wlmConnection={wlmConnection}
              additionalComments={additionalComments}
              textChange={this.textChange}
            />
            <Buttons handleSubmit={this.handleSubmit} />
          </CntrForm>
        </main>
      </MuiThemeProvider>
    );
  }
}

export default Form;
