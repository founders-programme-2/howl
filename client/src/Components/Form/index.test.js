import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Form from './index';

test('Natural', () => {
  const form = renderer.create(<BrowserRouter> <Form /> </BrowserRouter>).toJSON();
  expect(form).toMatchSnapshot();
});
