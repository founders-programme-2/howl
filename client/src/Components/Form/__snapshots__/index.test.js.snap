// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
Array [
  " ",
  <main>
    <header>
      <h1>
        Add to HOWL
      </h1>
    </header>
    <form
      className="sc-dnqmqq kCdHKN"
    >
      <h2>
        Contact Info
      </h2>
      <p>
        Our team of moderators will need to approve your story before it is published publicly. If we have any questions or concerns about your content, we will get in touch using your contact details. We will not use your details to send you spam mail.
      </p>
      <p
        className="sc-bdVaJa fKlUTO"
      />
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root ContactInfo-textField-1"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated ContactInfo-label-2"
          data-shrink={false}
          htmlFor="name"
        >
          Name (required)
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter your name or pseudo-name..."
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <p
        className="sc-bdVaJa fKlUTO"
      />
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root ContactInfo-textField-1"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated ContactInfo-label-2"
          data-shrink={false}
          htmlFor="email"
        >
          Email (required)
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter your email..."
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <p
        className="sc-bdVaJa fKlUTO"
      />
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root ContactInfo-textField-1"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated ContactInfo-label-2"
          data-shrink={false}
          htmlFor="phoneNum"
        >
          Phone Number (optional)
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="phoneNum"
            name="phone"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter your phone number. e.g:(+44 20 7946 0009)"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <h2>
        Your Contribution
      </h2>
      <p>
        Collecting personal stories, pictures and artifacts from real women is hugely important to us at HOWL. We value everyone's input and thank you in advance for sharing with us. We will now go through and ask you a series of questions about the story or image that you'd like to contribute. Don't worry if you can't remember something perfectly. Answer to the best of your ability and put any comments in 'Additional comments' section at the end of the form.
      </p>
      <p>
        Where did this event occur?
      </p>
      <div
        className="MuiFormControl-root Details-formControl-56"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Details-label-59"
          data-shrink={false}
          htmlFor="location"
        >
          Location (required)
        </label>
        <div
          aria-label="location"
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl Details-selectDropdown-55"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="select-location"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              id="location"
              name="location"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <p>
        Approximately when did this event occur? If the event spans multiple years, select the year when it started.
      </p>
      <div
        className="MuiFormControl-root Details-formControlDate-57"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Details-label-59"
          data-shrink={false}
          htmlFor="year"
        >
          Year (required)
        </label>
        <div
          aria-label="year"
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl Details-selectDropdown-55"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="select-year"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              id="year"
              name="year"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root Details-formControlDate-57 Details-formControlMonth-58"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Details-label-59"
          data-shrink={false}
          htmlFor="month"
        >
          Month (optional)
        </label>
        <div
          aria-label="month"
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl Details-selectDropdown-55"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="select-month"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              id="month"
              name="month"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <p>
        What kind of contribution would you like to make?
      </p>
      <div
        className="MuiFormControl-root Details-formControlRadio-60"
      >
        <legend
          className="MuiFormLabel-root"
        >
          I would like to share:
        </legend>
        <div
          className="MuiFormGroup-root"
          role="radiogroup"
        >
          <div
            className="MuiFormGroup-root MuiFormGroup-row"
          >
            <label
              className="MuiFormControlLabel-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-93 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-94 Mui-checked"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <div
                    className="PrivateRadioButtonIcon-root-109 PrivateRadioButtonIcon-checked-111"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-110"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                  <input
                    checked={true}
                    className="PrivateSwitchBase-input-96"
                    disabled={false}
                    name="storyType"
                    onChange={[Function]}
                    type="radio"
                    value="textPost"
                  />
                </span>
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
              >
                Text Post
              </span>
            </label>
            <label
              className="MuiFormControlLabel-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-93 MuiRadio-root MuiRadio-colorSecondary"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <div
                    className="PrivateRadioButtonIcon-root-109"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-110"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                  <input
                    checked={false}
                    className="PrivateSwitchBase-input-96"
                    disabled={false}
                    name="storyType"
                    onChange={[Function]}
                    type="radio"
                    value="imagePost"
                  />
                </span>
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
              >
                Image Post
              </span>
            </label>
            <label
              className="MuiFormControlLabel-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-93 MuiRadio-root MuiRadio-colorSecondary"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <div
                    className="PrivateRadioButtonIcon-root-109"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-110"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                  <input
                    checked={false}
                    className="PrivateSwitchBase-input-96"
                    disabled={false}
                    name="storyType"
                    onChange={[Function]}
                    type="radio"
                    value="both"
                  />
                </span>
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
              >
                Both
              </span>
            </label>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root Post-formControl-147"
      >
        <div
          className="sc-ifAKCX kPChof"
        >
          <p>
            What would you like to title your contribution to the archive (under 10 words)?
          </p>
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root Post-textField-142"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Post-label-144"
              data-shrink={false}
              htmlFor="title"
            >
              Title (required)
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="title"
                name="title"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Enter your Title..."
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
          <p>
            Please enter your story relating to the Women's Liberation Movement below. We encourage detailed memories but request you have a story that is longer than 1000 words, you email directly directly to us at fakeemail@howl.co.uk
          </p>
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root Post-textField-142 Post-textFieldDetails-143"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Post-label-144"
              data-shrink={false}
              htmlFor="details"
            >
              Details (required)
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
              onClick={[Function]}
            >
              <textarea
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                disabled={false}
                id="details"
                name="details"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Enter your story's details"
                required={false}
                rows="4"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <p>
        Please select one major category under which your contribution falls under. We understand that there might be more than one fits, please select one and you will be given a chance to select some subcategories later.
      </p>
      <div
        className="MuiFormControl-root AdditionalInfo-formControl-148"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated AdditionalInfo-label-152"
          data-shrink={false}
          htmlFor="category"
        >
          Category (required)
        </label>
        <div
          aria-label="category"
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl AdditionalInfo-select-149"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="select-category"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              id="category"
              name="category"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <p>
        Please select one or more 'tags' for your contribution. A tag can be thought of as a related theme or subcategory.
         
      </p>
      <div
        className="MuiFormControl-root AdditionalInfo-formControl-148 AdditionalInfo-formControlTags-150"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated AdditionalInfo-label-152"
          data-shrink={false}
          htmlFor="selectMultipleCheckbox"
        >
          Tags (required)
        </label>
        <div
          aria-label="tags"
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl AdditionalInfo-select-149"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              id="selectMultipleCheckbox"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <p>
        Tell us about yourself. What is your connection to the Women's Liberation Movement? Please do not share any private information here such as your phone number or address.
      </p>
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root AdditionalInfo-textField-151 AdditionalInfo-textFieldWlm-153"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated AdditionalInfo-label-152"
          data-shrink={false}
          htmlFor="standard-multiline-static"
        >
          Connection to Women’s Liberation Movement (required)
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            disabled={false}
            id="standard-multiline-static"
            name="wlmConnection"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="examples: activist, daughter of an activist, et cetera)"
            required={false}
            rows="4"
            value=""
          />
        </div>
      </div>
      <p>
        Anything else you would like to tell us?
      </p>
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root AdditionalInfo-textField-151"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated AdditionalInfo-label-152"
          data-shrink={false}
          htmlFor="additionalComments"
        >
          Additional comments? (optional)
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            disabled={false}
            id="additionalComments"
            name="additionalComments"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Talk to us!"
            required={false}
            rows="4"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root AdditionalInfo-formControlCheckbox-156"
      >
        <label
          className="MuiFormControlLabel-root AdditionalInfo-formControlLabel-157"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-93 MuiCheckbox-root MuiCheckbox-colorSecondary"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <input
                checked={false}
                className="PrivateSwitchBase-input-96"
                data-indeterminate={false}
                disabled={false}
                onChange={[Function]}
                type="checkbox"
                value="infoTrue"
              />
            </span>
          </span>
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
          >
            I certify that this information is true to the best of my knowledge and contains no personal attacks.
          </span>
        </label>
      </div>
      <div
        className="sc-gqjmRU ecJXIY"
      >
        <a
          href="/add"
          onClick={[Function]}
        >
          <input
            className="sc-gZMcBi kZNDNB"
            type="button"
            value="Go Back"
          />
        </a>
        <input
          className="sc-iwsKbI leKRwC"
          onClick={[Function]}
          type="button"
          value="Submit"
        />
      </div>
    </form>
  </main>,
  " ",
]
`;
